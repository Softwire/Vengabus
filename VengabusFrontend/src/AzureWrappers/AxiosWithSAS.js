import axios from 'axios';
import moment from 'moment';
import crypto from 'crypto';


/**
 * Provides methods to interact with the api using an SAS token for authentication.
 */
export default class AxiosWithSAS {
    /**
     * @param {string} connectionString The connection string of the server to connect to.
     */
    constructor(connectionString) {
        this.connectionString = connectionString;
    }

    /**
     * Performs a GET request after being passed a url, with the appropriate SAS header.
     * @param {string} url The url that the get request should be sent to.
     * @param {string} config An axios config that is used in the axios request in addition to the SAS authentication.
     * @returns {Promise} The Promise returned by the axios request.
     */
    get = (url, config) => {
        const token = this.getTokenFromConnectionString();
        const sasConfig = this.addAuthToConfig(config, token);
        return axios.get(url, sasConfig);
    }

    /**
     * Performs a DELETE request after being passed a url, with the appropriate SAS header.
     * @param {string} url The url that the get request should be sent to.
     * @param {string} config An axios config that is used in the axios request in addition to the SAS authentication.
     * @returns {Promise} The Promise returned by the axios request.
     */
    delete = (url, config) => {
        const token = this.getTokenFromConnectionString();
        const sasConfig = this.addAuthToConfig(config, token);
        return axios.delete(url, sasConfig);
    }

    /**
     * Performs a PUSH request after being passed a url, with the appropriate SAS header.
     * @param {string} url The url that the get request should be sent to.
     * @param {string} config An axios config that is used in the axios request in addition to the SAS authentication.
     * @returns {Promise} The Promise returned by the axios request.
     */
    push = (url, config) => {
        const token = this.getTokenFromConnectionString();
        const sasConfig = this.addAuthToConfig(config, token);
        return axios.push(url, sasConfig);
    }

    /**
     * Returns an SAS token valid for 1 hour based on the stored connectionString.
     * @returns {object} The SAS token.
     */
    getTokenFromConnectionString() {
        const connectionArray = this.connectionString.split(';');
        let endPoint, sharedAccessKeyName, sharedAccessKey;

        for (let i = 0; i < connectionArray.length; i++) {
            const currentSection = connectionArray[i];

            if (currentSection.startsWith('Endpoint=')) {
                endPoint = 'https:/' + currentSection.slice(13);
            }
            if (currentSection.startsWith('SharedAccessKeyName=')) {
                sharedAccessKeyName = currentSection.slice(20);
            }
            if (currentSection.startsWith('SharedAccessKey=')) {
                sharedAccessKey = currentSection.slice(16);
            }
        }
        return this.createSasToken(endPoint, sharedAccessKeyName, sharedAccessKey);
    }

    /**
     * Generates an SAS token from a given uri, key name, and key.
     * @param {string} uri The URI of the azure service bus.
     * @param {string} keyName The name of the key used to connect to the server, usually "RootManageSharedAccessKey"
     * @param {string} key The key used to connect to the server.
     * @returns {object} The SAS token, valid for 1 hour.
     */
    createSasToken(uri, keyName, key) {
        // Token expires in one hour
        const expiry = moment().add(1, 'hours').unix();

        const string_to_sign = encodeURIComponent(uri) + '\n' + expiry;
        var hmac = crypto.createHmac('sha256', key);
        hmac.update(string_to_sign);
        const signature = hmac.digest('base64');
        const token = 'SharedAccessSignature sr=' + encodeURIComponent(uri) + '&sig=' + encodeURIComponent(signature) + '&se=' + expiry + '&skn=' + keyName;

        return token;
    }

    /**
     * Adds the given SAS token to the headers of the given axios config.
     * @param {object} config The existing axios config.
     * @param {object} token The SAS token to be put in the headers.
     * @returns {object} The config object.
     */
    addAuthToConfig(config, token) {
        const extendedConfig = config || {};
        extendedConfig.headers = extendedConfig.headers || {};
        extendedConfig.headers["Auth-SAS"] = token;
        return extendedConfig;
    }


}