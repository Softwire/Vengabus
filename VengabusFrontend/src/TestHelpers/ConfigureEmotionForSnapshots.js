//import * as selectorMatchers from 'jest-enzyme-selector-matchers';
import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion';
import Adaptor from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
configure({ adapter: new Adaptor() });
expect.addSnapshotSerializer(createSerializer(emotion));
//expect.extend(selectorMatchers);u