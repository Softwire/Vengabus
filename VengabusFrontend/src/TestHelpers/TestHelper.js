import { Button } from "react-bootstrap";

class TestHelper {
    /*
     * It returns a promise that waits until the state of the components is updated and then continues.
     * 
     * Expected usage is: 
     *     return testHelper.afterReactHasUpdated().then(() => {
     *         expect('actualValue').toEqual('expectedValue');
     *     });
     * 
     * Notes on multi-layered async tests (e.g. wait for this, and then wait for that, and then expect a result) will be forthcoming.
     */
    afterReactHasUpdated(noArgsGiven) {
        if (noArgsGiven) {
            throw new Error('afterReactHasUpdated() does not expect to take any arguments. See expected usage for further details.');
        }

        return new Promise(function (resolve) {
            setTimeout(resolve, 0);
        });
    }
    clickElementWithId(wrapper, id) {
        wrapper.find(id).at(0).simulate("click");
    }
}

export const testHelper = new TestHelper();
