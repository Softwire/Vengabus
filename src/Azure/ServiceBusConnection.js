let azure = require('azure-sb');

class ServiceBusConnection {
    constructor() {
        this.activeServiceBusConString = '';
        this.activeQueueName = '';
        this.useHardCodedDirectConString = true;
        this.useHardCodedIndirectConString = false;
    }

    setConnectionString = (newConnectionString) => {
        this.activeServiceBusConString = newConnectionString;
    };

    setQueueName = (newQueueName) => {
        this.activeQueueName = newQueueName;
    };

    getServiceBusService = () => {
        this.applyHardCodedOverrides();
        try {
            let serviceBusService = azure.createServiceBusService(this.activeServiceBusConString);
            return serviceBusService;
        } catch (err) {
            console.log(err);
            return;
        }
    };

    applyHardCodedOverrides() {
        // Current temporary code, until we've setup sensible 'caching' of these values.
        // Override settings to point at known valid target:
        let directConString =
            'Endpoint=sb://vengabusdemo.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=a8sPg0N79mgTNAfM57b7kwWsBCPK8TW7hwTwyexlK+8=';
        let directQueueName = 'DemoQueue1';

        // Further override settings to point at Rev.Proxy. target:
        let indirectConString =
            'Endpoint=sb://vengabusreverseproxy.azurewebsites.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=a8sPg0N79mgTNAfM57b7kwWsBCPK8TW7hwTwyexlK+8=';
        let indirectQueueName = 'AzureApiProxy/' + directQueueName;

        if (this.useHardCodedDirectConString) {
            this.activeServiceBusConString = directConString;
            this.activeQueueName = directQueueName;
        }

        if (this.useHardCodedIndirectConString) {
            this.activeServiceBusConString = indirectConString;
            this.activeQueueName = indirectQueueName;
        }
    }
}
export const serviceBusConnection = new ServiceBusConnection();
