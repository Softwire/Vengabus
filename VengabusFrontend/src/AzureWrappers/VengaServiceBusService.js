import { AxiosWithSAS } from './AxiosWithSAS';
require('util.promisify').shim();

/*
    This class exists as our own wrapper around the equivalent Azure API object,
    so that we can make any appropriate improvements to its API.
    e.g. Converting everything to promises.  
*/
export class VengaServiceBusService {
    constructor(connectionString, apiRoot) {
        this.axiosWithSAS = new AxiosWithSAS(connectionString);
        this.apiRoot = apiRoot;
        this.jsonConfig = {
            headers: {
                "Content-Type": "application/json"
            }
        };
    }

    /**
     * Gets the details of all queues in the current namespace from the server.
     * @return {object} The queues returned by the server.
     */
    listQueues = () => {
        const url = this.apiRoot + 'queues';
        return this.axiosWithSAS.get(url);
    }

    /**
     * Gets the details of a particular queue.
     * @param {string} queueName The name of the queue to get details for.
     * @return {object} The queues returned by the server.
     */
    getQueueDetails = (queueName) => {
        const url = this.apiRoot + `queues/${queueName}`;
        return this.axiosWithSAS.get(url);
    }

    /**
     * Updates the properties of a queue
     * @param {object} queueDescription Object containing the new properties of the queue. It is specified in this which queue we are updating.
     * For more information on the form of this object see https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.queuedescription?view=azure-dotnet
     */
    updateQueue = (queueDescription) => {
        const url = this.apiRoot + 'queues/update';
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, queueDescription, config);
    }

    /**
     * Renames a queue
     * @param {string} oldName The original name of the queue.
     * @param {string} newName The new name of the queue.
     */
    renameQueue = (oldName, newName) => {
        const url = this.apiRoot + 'queues/rename';
        const config = this.jsonConfig;
        const names = {
            oldName: oldName,
            newName: newName
        };
        this.axiosWithSAS.post(url, names, config);
    }

    /**
     * Deletes a queue
     * @param {string} name Name of the queue to be deleted.
     */
    deleteQueue = (name) => {
        const url = this.apiRoot + 'queues/delete/' + name;
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, null, config);
    }

    /**
     * Gets the details of all topics in the current namespace from the server.
     * @return {object} The queue returned by the server.
     */
    listTopics = () => {
        const url = this.apiRoot + 'topics';
        return this.axiosWithSAS.get(url);
    }

    /**
     * Gets the details of a particular topic.
     * @param {string} topicName The name of the topic to get details for.
     * @return {object} The topic returned by the server.
     */
    getTopicDetails = (topicName) => {
        const url = this.apiRoot + `topics/${topicName}`;
        return this.axiosWithSAS.get(url);
    }

    /**
     * Updates the properties of a queue
     * @param {object} topicDescription Object containing the new properties of the topic. It is specified in this which topic we are updating.
     * For more information on the form of this object see https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.topicdescription?view=azure-dotnet
     */
    updateTopic = (topicDescription) => {
        const url = this.apiRoot + 'topics/update';
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, topicDescription, config);
    }

    /**
     * Renames a topic
     * @param {string} oldName The original name of the topic.
     * @param {string} newName The new name of the topic.
     */
    renameTopic = (oldName, newName) => {
        const url = this.apiRoot + 'topics/rename';
        const config = this.jsonConfig;
        const names = {
            oldName: oldName,
            newName: newName
        };
        this.axiosWithSAS.post(url, names, config);
    }

    /**
     * Deletes a topic
     * @param {string} name Name of the topic to be deleted.
     */
    deleteTopic = (name) => {
        const url = this.apiRoot + 'topics/delete/' + name;
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, null, config);
    }

    /**
     * Gets the details of all subscriptions in a given topic from the server.
     * @param {string} topicName The name of the topic to get subscriptions from.
     * @return {object} The subsctiptions returned by the server.
     */
    listSubscriptions = (topicName) => {
        const url = this.apiRoot + `subscriptions/${topicName}`;
        return this.axiosWithSAS.get(url);
    }


    /**
     * Gets the details of a particular topic.
     * @param {string} parentTopicName The name of the parent topic for the subscription.
     * @param {string} subscriptionName The name of the subscription to get details for.
     * @return {object} The topic returned by the server.
     */
    getSubscriptionDetails = (parentTopicName, subscriptionName) => {
        const url = this.apiRoot + `subscriptions/${parentTopicName}/${subscriptionName}`;
        return this.axiosWithSAS.get(url);
    }

    /**
     * Updates the properties of a subscription
     * @param {object} subDescription Object containing the new properties of the subscription. It is specified in this which subscription we are updating.
     * For more information on the form of this object see https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.subscriptiondescription?view=azure-dotnet
     */
    updateSubscription = (subDescription) => {
        const url = this.apiRoot + 'subscriptions/update';
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, subDescription, config);
    }

    /**
     * Deletes a subscription
     * @param {string} name Name of the subscription to be deleted.
     * @param {string} parentTopic Name of the paretn topic of the subscription.
     */
    deleteSubscription = (name, parentTopic) => {
        const url = this.apiRoot + 'subscriptions/delete/' + name;
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, parentTopic, config);
    }

    /**
     * Sends a message to the given queue on the server.
     * @param {string} queueName The name of the queue to send the message to.
     * @param {object} message The message to send to the queue, in the format:
     * {
     *      "customProperties": {},
     *      "predefinedProperties": {},
     *      "messageBody": "string"
     * }
     */
    sendMessageToQueue = (queueName, message) => {
        const url = this.apiRoot + `queues/${queueName}/messages`;
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, message, config);
    }

    /**
     * Sends a message to the given topic on the server.
     * @param {string} topicName The name of the topic to send the message to.
     * @param {object} message The message to send to the queue, in the format:
     * {
     *      "customProperties": {},
     *      "predefinedProperties": {},
     *      "messageBody": "string"
     * }
     */
    sendMessageToTopic = (topicName, message) => {
        const url = this.apiRoot + `topics/${topicName}/messages`;
        const config = this.jsonConfig;
        this.axiosWithSAS.post(url, message, config);
    }

    purgeQueueMessages = (queueName) => {
        const url = this.apiRoot + `queues/${queueName}/messages`;
        return this.axiosWithSAS.delete(url);
    }

    deleteQueueSingleMessage = (queueName, messageId, uniqueId) => {
        const messageIdEncoded = encodeURIComponent(messageId);
        const url = this.apiRoot + `queues/${queueName}/messages/${uniqueId}?messageId=${messageIdEncoded}`;
        return this.axiosWithSAS.delete(url);
    }

    deleteQueueSingleDeadLetterMessage = (queueName, messageId, uniqueId) => {
        const messageIdEncoded = encodeURIComponent(messageId);
        const url = this.apiRoot + `queues/${queueName}/deadletters/${uniqueId}?messageId=${messageIdEncoded}`;
        return this.axiosWithSAS.delete(url);
    }

    purgeTopicMessages = (topicName) => {
        const url = this.apiRoot + `topics/${topicName}/messages`;
        return this.axiosWithSAS.delete(url);
    }

    deleteTopicSingleMessage = (topicName, messageId, uniqueId) => {
        const messageIdEncoded = encodeURIComponent(messageId);
        const url = this.apiRoot + `topics/${topicName}/messages/${uniqueId}?messageId=${messageIdEncoded}`;
        return this.axiosWithSAS.delete(url);
    }

    purgeSubscriptionMessages = (topicName, subscriptionName) => {
        const url = this.apiRoot + `subscriptions/${topicName}/${subscriptionName}/messages`;
        return this.axiosWithSAS.delete(url);
    }

    deleteSubscriptionSingleMessage = (topicName, subscriptionName, messageId, uniqueId) => {
        const messageIdEncoded = encodeURIComponent(messageId);
        const url = this.apiRoot + `subscriptions/${topicName}/${subscriptionName}/messages/${uniqueId}?messageId=${messageIdEncoded}`;
        return this.axiosWithSAS.delete(url);
    }

    deleteSubscriptionSingleDeadLetterMessage = (topicName, subscriptionName, messageId, uniqueId) => {
        const messageIdEncoded = encodeURIComponent(messageId);
        const url = this.apiRoot + `subscriptions/${topicName}/${subscriptionName}/deadletters/${uniqueId}?messageId=${messageIdEncoded}`;
        return this.axiosWithSAS.delete(url);
    }

    listQueueMessages = (queueName, messageCount) => {
        let queryString = "";
        if (messageCount) {
            queryString = "/?messageCount=" + messageCount;
        }
        const url = this.apiRoot + `queues/${queueName}/messages` + queryString;
        return this.axiosWithSAS.get(url);
    }

    listSubscriptionMessages = (topicName, subscriptionName, messageCount) => {
        let queryString = "";
        if (messageCount) {
            queryString = "/?messageCount=" + messageCount;
        }
        const url = this.apiRoot + `subscriptions/${topicName}/${subscriptionName}/messages` + queryString;
        return this.axiosWithSAS.get(url);
    }


    listQueueDeadLetterMessages = (queueName, messageCount) => {
        let queryString = "";
        if (messageCount) {
            queryString = "/?messageCount=" + messageCount;
        }
        const url = this.apiRoot + `queues/${queueName}/deadletters` + queryString;
        return this.axiosWithSAS.get(url);
    }

    listSubscriptionDeadLetterMessages = (topicName, subscriptionName, messageCount) => {
        let queryString = "";
        if (messageCount) {
            queryString = "/?messageCount=" + messageCount;
        }
        const url = this.apiRoot + `subscriptions/${topicName}/${subscriptionName}/deadletters` + queryString;
        return this.axiosWithSAS.get(url);
    }

    getQueueMostRecentDeadletter = (queueName) => {
        const url = this.apiRoot + `queues/${queueName}/mostRecentDeadletter`;
        return this.axiosWithSAS.get(url);
    }

    getSubscriptionMostRecentDeadletter = (topicName, subscriptionName) => {
        const url = this.apiRoot + `subscriptions/${topicName}/${subscriptionName}/mostRecentDeadletter`;
        return this.axiosWithSAS.get(url);
    }



    /**
     * Returns the pre-defined properties that can be read from messages.
     * @returns {string[]} The available properties.
     */
    getReadableMessageProperties = () => {
        const url = this.apiRoot + 'messages/properties/readable';
        return this.axiosWithSAS.get(url);
    }

    /**
     * Returns the pre-defined properties that can be set to messages.
     * @returns {string[]} The allowed properties.
     */
    getWriteableMessageProperties = () => {
        const url = this.apiRoot + 'messages/properties/writeable';
        return this.axiosWithSAS.get(url);
    }
}
