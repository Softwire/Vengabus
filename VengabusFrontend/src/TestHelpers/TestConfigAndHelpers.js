import './ConfigureEnzyme';
import { testHelper as helper } from './TestHelper';

export const testHelper = helper;
