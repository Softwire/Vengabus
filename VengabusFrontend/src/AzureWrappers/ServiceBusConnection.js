import { VengaServiceBusService } from './VengaServiceBusService';

class ServiceBusConnection {
    constructor() {
        this.activeServiceBusConString = '';
        this.activeAPIroot = '';
    }

    /**
     * Updates the Azure connection string used to connect to the service bus.
     * @param {string} newConnectionString The new connection string to use.
     * Should be in the format generated by the Azure portal:
     * Endpoint=sb://<EndpointLocation>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<SharedAccessKey>
     * e.g.
     * Endpoint=sb://vengabusdemo.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SGd6kBFkdDBGKs5FIBSgibSgSG5dsUBGS+8=
     */
    setConnectionString = (newConnectionString) => {
        this.activeServiceBusConString = newConnectionString;
    };

    /**
     * Updates the location of the API backend to be used by the ServiceBusConnection.
     * http:// and a trailing slash will be added if omitted.
     * @param {string} newURI The new API root URI to use.
     */
    setApiRoot = (newURI) => {
        this.activeAPIroot = this.formatApiRootString(newURI);
    };

    /**
     * Takes a string and prepends http:// and adds a trailing slash if not already present.
     * @param {string} APIRootString The string to format.
     * @return {string} The formatted string.
     */
    formatApiRootString = (APIRootString) => {
        if (!APIRootString.startsWith("http")) { //Catches both http and https cases
            APIRootString = "http://" + APIRootString;
        }
        if (!APIRootString.endsWith("/")) {
            APIRootString = APIRootString + "/";
        }
        return APIRootString;
    };

    /**
     * Returns a VengaServiceBusService that uses the current connection string and API root location.
     * @return {VengaServiceBusService} The VengaServiceBusService defined by the settings in this object.
     */
    getServiceBusService = () => {
        try {
            return new VengaServiceBusService(this.activeServiceBusConString, this.activeAPIroot);
        } catch (err) {
            // Obviously we'd prefer this didn't happen, but it can easily occur if, e.g. ConnStrings aren't provided yet.
            // Failing to create a connection shouldn't be an immediate fatal error ... but *USING* a failed connection probably should.
            // So don't error - just return `undefined`.
            console.error(err);
            return undefined;
        }
    };
}
export const serviceBusConnection = new ServiceBusConnection();
