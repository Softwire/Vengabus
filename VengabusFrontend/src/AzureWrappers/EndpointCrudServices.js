import { serviceBusConnection } from './ServiceBusConnection';

/*
  Common Interface:
  {
    rename: function(oldName, newName),
    update: function(endpointData),
    delete: function(name)
  }
 */

export class QueueCrudService {
    constructor() {
        this.serviceBusService = serviceBusConnection.getServiceBusService();

        this.rename = this.serviceBusService.renameQueue;
        this.update = this.serviceBusService.updateQueue;
        this.delete = this.serviceBusService.deleteQueue;
    }
}

export class TopicCrudService {
    constructor() {
        this.serviceBusService = serviceBusConnection.getServiceBusService();

        this.rename = this.serviceBusService.renameTopic;
        this.update = this.serviceBusService.updateTopic;
        this.delete = this.serviceBusService.deleteTopic;
    }
}

export class SubscriptionCrudService {
    constructor(parentTopic) {
        this.serviceBusService = serviceBusConnection.getServiceBusService();
        this.parentTopic = parentTopic;

        this.rename = () => console.error('cannot rename subscriptions because #Microsoft');
        this.update = this.serviceBusService.updateSubscription;
        this.delete = (subscriptionName) => this.serviceBusService.deleteSubscription(subscriptionName, this.parentTopic);
    }
}
