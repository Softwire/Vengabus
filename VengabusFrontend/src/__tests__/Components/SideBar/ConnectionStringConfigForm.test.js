import 'jest-localstorage-mock';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { testHelper } from '../../../TestHelpers/TestHelper';
import React from 'react';
import {
    ConnectionStringConfigForm,
    LOCAL_STORAGE_STRINGS
} from "../../../Components/SideBar/ConnectionStringConfigForm";
import { serviceBusConnection } from '../../../AzureWrappers/ServiceBusConnection';

it('component renders fine when connection string localStorage is not present', () => {
    localStorage.setItem(LOCAL_STORAGE_STRINGS.ConnectionString, undefined);
    const wrapper = mount(<ConnectionStringConfigForm />);
});

it('component renders fine when connection string localStorage *is* present', () => {
    localStorage.setItem(LOCAL_STORAGE_STRINGS.ConnectionString, 'some Value');
    const wrapper = mount(<ConnectionStringConfigForm />);
});

it('localStore is updated when the connection string form is changed', () => {
    localStorage.setItem(LOCAL_STORAGE_STRINGS.ConnectionString, 'before');
    const wrapper = mount(<ConnectionStringConfigForm />);

    //Cannot use the id of the form to find it because the wrapper then contains two elements instead of one
    //Not sure why this happens
    const connectionStringInput = wrapper.find('input[placeholder="Enter Connection String"]');
    connectionStringInput.simulate('change', { target: { value: 'after' } });

    return testHelper.afterReactHasUpdated().then(() => {
        expect(localStorage.getItem(LOCAL_STORAGE_STRINGS.ConnectionString)).toEqual('after');
    });
});

it('localStore is updated when the API location form is changed', () => {
    localStorage.setItem(LOCAL_STORAGE_STRINGS.APIroot, 'before');
    const wrapper = mount(<ConnectionStringConfigForm />);

    //Cannot use the id of the form to find it because the wrapper then contains two elements instead of one
    //Not sure why this happens
    const connectionStringInput = wrapper.find('#APILocationForm');
    connectionStringInput.simulate('change', { target: { value: 'after' } });

    return testHelper.afterReactHasUpdated().then(() => {
        expect(localStorage.getItem(LOCAL_STORAGE_STRINGS.APIroot)).toEqual('after');
    });
});

it('ServiceBusConnection is updated from LocalStorage on page load', () => {
    serviceBusConnection.setConnectionString('connStringInConnection');
    localStorage.setItem(LOCAL_STORAGE_STRINGS.ConnectionString, 'connStringInLocalStorage');
    serviceBusConnection.setApiRoot('APIRootInConnection');
    localStorage.setItem(LOCAL_STORAGE_STRINGS.APIroot, 'http://APIRootInLocalStorage/');

    mount(<ConnectionStringConfigForm />);
    return testHelper.afterReactHasUpdated().then(() => {
        expect(serviceBusConnection.activeServiceBusConString).toEqual('connStringInLocalStorage');
        expect(serviceBusConnection.activeAPIroot).toEqual('http://APIRootInLocalStorage/');
    });
});

it('API root location is formatted correctly', () => {
    serviceBusConnection.setApiRoot('APIRootInConnection');
    localStorage.setItem(LOCAL_STORAGE_STRINGS.APIroot, 'UnformattedAPIRootInLocalStorage');

    mount(<ConnectionStringConfigForm />);
    return testHelper.afterReactHasUpdated().then(() => {
        expect(serviceBusConnection.activeAPIroot).toEqual('http://UnformattedAPIRootInLocalStorage/');
    });
});

it('ConnectionStringConfigForm renders correctly', () => {
    let configForm = renderer.create(<ConnectionStringConfigForm />);
    expect(configForm.toJSON()).toMatchSnapshot();
});