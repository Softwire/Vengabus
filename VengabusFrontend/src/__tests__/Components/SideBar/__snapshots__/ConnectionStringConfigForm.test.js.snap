// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStringConfigForm renders correctly 1`] = `
.emotion-10 {
  margin: 0 5px 0 0;
}

.emotion-10.btn {
  padding: 6px 13px;
}

.emotion-11 {
  margin: 0 0 0 5px;
}

.emotion-11.btn {
  padding: 6px 13px;
}

.emotion-15 {
  padding: 5px;
}

.emotion-9 {
  color: black;
}

.emotion-8 {
  position: relative;
  box-sizing: border-box;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0,0%,98%);
  border-color: hsl(0,0%,80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 38px;
  outline: 0 !important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  box-sizing: border-box;
}

.emotion-7:hover {
  border-color: hsl(0,0%,70%);
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  box-sizing: border-box;
}

.emotion-0 {
  color: #222;
  margin-left: 2px;
  margin-right: 2px;
  max-width: calc(100% - 8px);
  overflow: hidden;
  position: absolute;
  text-overflow: ellipsis;
  white-space: nowrap;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  box-sizing: border-box;
}

.emotion-1 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: #222;
  box-sizing: border-box;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-3 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: hsl(0,0%,80%);
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
}

.emotion-5 {
  color: hsl(0,0%,80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-5:hover {
  color: hsl(0,0%,60%);
}

.emotion-4 {
  display: inline-block;
  fill: currentColor;
  line-height: 1;
  stroke: currentColor;
  stroke-width: 0;
}

.emotion-12 {
  cursor: not-allowed;
  float: right;
}

.emotion-12 button {
  pointer-events: auto;
}

.emotion-14 {
  color: black;
  overflow-wrap: break-word;
  max-height: calc(100vh - 275px - 50px);
  overflow-y: auto;
}

.emotion-13 {
  background: #66b3ff;
}

<form
  className="emotion-15"
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="connectionStringLabel"
    >
      Select Existing/Create New String
    </label>
    <div
      className="emotion-8 emotion-9"
      id={undefined}
      onKeyDown={[Function]}
    >
      <div
        className="emotion-7"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-0"
          >
            
          </div>
          <div
            className="emotion-1"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                aria-label={undefined}
                aria-labelledby={undefined}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className={undefined}
                disabled={false}
                id="react-select-8-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="emotion-6"
        >
          <span
            className="emotion-3"
          />
          <div
            aria-hidden="true"
            className="emotion-5"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="emotion-4"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group has-error"
  >
    <label
      className="control-label"
      htmlFor="connectionString"
    >
      Connection String
    </label>
    <input
      className="form-control"
      id="connectionString"
      onChange={[Function]}
      placeholder="Enter Connection String"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="APILocationForm"
    >
      ServiceBus API Server String
    </label>
    <input
      className="form-control"
      id="APILocationForm"
      onChange={[Function]}
      placeholder="API Location, e.g. http://a.com:63591/"
      type="text"
      value="UnformattedAPIRootInLocalStorage"
    />
  </div>
  <button
    className="emotion-10 btn btn-default"
    disabled={false}
    id="connectButton"
    onClick={[Function]}
    type="button"
  >
    Connect
  </button>
  <div
    className="emotion-12"
    onBlur={[Function]}
    onClick={null}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <button
      className="emotion-11 btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete Connection
    </button>
  </div>
  <div>
    <br />
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="emotion-13 panel-heading"
    >
      Connection String
    </div>
    <div
      className="emotion-14 panel-body"
    >
      <div>
        Connected to:  
      </div>
      <div>
        Connection String:  
      </div>
    </div>
  </div>
</form>
`;
